### ISSUE TITLE
Explore Metrics: Raw Prometheus Container needs to not overwrite mixed datasource results for display

### ISSUE THREAD

**Issue by gelicia** [grafana/grafana#75036]
### What happened?

The raw prometheus container will get data from a table result or a raw prometheus result, with the table result overriding ([src](https://github.com/grafana/grafana/blob/main/public/app/features/explore/RawPrometheus/RawPrometheusContainer.tsx#L38))

In the case a mixed datasource returns both table result and raw prometheus result data, we will want to be sure the raw prometheus data is not lost.

### What did you expect to happen?

For the prometheus data to show in the raw prometheus view, and the loki data to show in the table view.

### Did this work before?

N/A

### How do we reproduce it?

1. Run Devenv, `sources=tempo` - you will only need loki and prometheus for this
2. In Explore, create a mixed datasource query
3. Create a loki query, `sum by(status) (rate({compose_service="app"} | logfmt [$__auto]))`
4. Create a prometheus query `sum by(code) (prometheus_http_requests_total{job="prometheus"})`
5. Ensure both queries are of type "range"
6. Run, see the results are merged, with a timeseries chart showing both "status" (from loki) and "code" from prometheus
7. Change both queries to "instant" and run
8. See the raw prometheus panel with "status" results (from loki)
9. See the table panel with "status" results (from loki)
10. Note the prometheus "code" results do not show

### Environment (with versions)?

Grafana: Main
OS:
Browser:

### Grafana platform?

Other

### Datasource(s)?

Prometheus and Loki

**Comment by itsmylife** [comment]
@gelicia Could you please provide a https://play.grafana.org/ link where we can reproduce and see the issue?

**Comment by gelicia** [comment]
@itsmylife is there a reason the replication steps provided are not sufficient? they will work using devenv

**Comment by itsmylife** [comment]
@gelicia Thank you for the additional explanation. That helped a lot. 
We have a clear way of reproducing the issue. We will take a look at this.

**Comment by itsmylife** [comment]
@grafana/observability-metrics does not currently have the bandwidth to prioritize development on this issue. If members of the community want to take this issue we will help review any submitted PRs. Please upvote or comment to help us re-prioritize this issue in the future!

**Comment by gelicia** [comment]
@grafana/observability-metrics Can you confirm if https://github.com/grafana/grafana/issues/81461 is a duplicate of this issue? I know the replication steps are a little different, but it's the same thing where the prometheus results are overwritten.

**Comment by itsmylife** [comment]
@gelicia Yes it is the same problem. Thanks for linking those. I'll try to allocate some time and try to fix this in this quarter.
